validator-auction-rest-server/lib/EthEventsClient.js

```diff
@@ -19,31 +19,37 @@ export default class EthEventsClient {
     }
 
     async getAuctionSummary() {
-        const result = {
-            contractAddress: this._contractAddress
-        }
-        const [currentBlockTime, bids, auctionStart, whitelistedAddresses, deploymentParams] = await Promise.all([
-            this.getCurrentBlockTime(),
-            this.getBidEvents(),
-            this.getAuctionStartInSeconds(),
-            this.getWhitelistedAddresses(),
-            this.getAuctionDeploymentParameters()
-        ])
-
-        if(!deploymentParams || !auctionStart) {
-            result.auctionHasNotStarted = true
-            return result
+        const [currentBlockTime, allEvents] = await Promise.all([this.getCurrentBlockTime(), this.getAllEvents()])
+
+        const state = this.getAuctionState(allEvents)
+
+        if (state === 'Not Deployed') {
+            return {
+                auctionState: state
+            }
         }
 
-        result.whitelistedAddresses = whitelistedAddresses
-        result.bids = bids
-        result.takenSlotsCount = bids.length
-        result.freeSlotsCount = deploymentParams.numberOfParticipants - result.takenSlotsCount
-        result.remainingSeconds = EthEventsClient.calculateRemainingAuctionSeconds(auctionStart, currentBlockTime, deploymentParams.durationInDays)
-        result.currentPriceInWEI = EthEventsClient.getCurrentPriceAsBigNumber(auctionStart * 1000, currentBlockTime * 1000, deploymentParams.durationInDays, deploymentParams.startPrice).toString()
-        result.priceFunction = EthEventsClient.calculateAllSlotPrices(auctionStart, deploymentParams.durationInDays, deploymentParams.startPrice)
-        result.currentBlocktimeInMs = currentBlockTime * 1000
-        return result
+        const deploymentParams = this.getAuctionDeploymentParameters(allEvents)
+        const bids = this.getBids(allEvents)
+        const whitelistedAddresses = this.getWhitelistedAddresses(allEvents)
+        let auctionStart = this.getAuctionStartInSeconds(allEvents)
+        if (!auctionStart) {
+            auctionStart = currentBlockTime
```

Comment:
What is this, why do we take the currentBlockTime as starting time if the auction did not start yet?

Reply:
The currentBlockTime is not rendered anywhere on the UI, it is just used for calculation of the price curve.

Reply:
The auction price is only meaningful if the auction is started, so we should only calculate it if the state it started. This is connected to my other comment then I guess.